dissect/target/helpers/fsutil.py

```diff
@@ -289,6 +290,18 @@ def walk_ext(path_entry, topdown=True, onerror=None, followlinks=False):
         yield [path_entry], dirs, files
 
 
+def walk_ng(path_entry: filesystem.FilesystemEntry) -> Iterator[filesystem.FilesystemEntry]:
+    """Recursively walk the given FilesystemEntry, yields FilesystemEntry instances."""
+    if not path_entry.is_dir():
+        return
+
+    for child_entry in path_entry.scandir():
+        yield child_entry
+
+        if child_entry.is_dir() and not child_entry.is_symlink():
+            yield from walk_ng(child_entry)
```

Comment:
```suggestion
    yield path_entry
    
    if not path_entry.is_dir():
        return

    for child_entry in path_entry.scandir():
    
        if child_entry.is_dir() and not child_entry.is_symlink():
            yield from walk_ng(child_entry)
        else:
            yield child_entry
```

This should return the same results for the walkfs plugin (avoid the duplicate outputs) I think.